File: src/main/structs/Language.java
package structs;

// Literally just a struct for the languages
public class Language {
    // Store the language name
    public String langName;
    // Store the spoken skill level
    public String spoken;
    // Store the written skill level
    public String written;
    // Store whether the language is the user's native language or not
    public boolean nativeLang;

    // Constructors
    public Language() {
        // Initialize variables
        this.langName = "";
        this.spoken = "";
        this.written = "";
        this.nativeLang = false;
    }
}

File: src/main/structs/Language.kt
package structs

class Language {
    @JvmField
    var langName: String
    @JvmField
    var spoken: String
    @JvmField
    var written: String
    @JvmField
    var nativeLang: Boolean

    constructor(langName: String, spoken: String, written: String, nativeLang: Boolean) {
        this.langName = langName
        this.spoken = spoken
        this.written = written
        this.nativeLang = nativeLang
    }

    constructor() {
        langName = ""
        spoken = ""
        written = ""
        nativeLang = false
    }
}